src/kernel/sporadic.c

```diff
@@ -238,44 +238,23 @@ void refill_budget_check(ticks_t usage)
 {
     sched_context_t *sc = NODE_STATE(ksCurSC);
     /* this function should only be called when the sc is out of budget */
-    ticks_t capacity = refill_capacity(NODE_STATE(ksCurSC), usage);
```

Comment:
I'm least certain about this change. I'm not going to block merging it but it would be very nice to prove some properties about it.

Reply:
I removed the commit for now.